bqskit/passes/util/blockanalysis.py

```diff
@@ -0,0 +1,80 @@
+"""This module implements the BlockAnalysisPass."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+from typing import Callable
+
+from bqskit.compiler.basepass import BasePass
+from bqskit.ir.circuit import Circuit
+from bqskit.ir.gate import Gate
+from bqskit.ir.gates import CircuitGate
+
+_logger = logging.getLogger(__name__)
+
+
+def gate_count(block: Circuit) -> dict[Gate, int]:
+    return block.gate_counts
+
+
+class BlockAnalysisPass(BasePass):
+    """
+    Scaffold to analyze blocks of type CircuitGate.
+
+    A Circuit may be partitioned into blocks, each of which may have very
+    different properties. This pass allows for blocks in a circuit to be
+    analyzed and compared quickly.
+
+    This pass scans through a circuit, and stores the results of the scan in a
+    list variable called `results`.
+    """
+
+    def __init__(
+        self,
+        analysis_function: Callable[[Circuit], Any] = gate_count,
+        filter_function: Callable[[Circuit], Any] = lambda x: x,
```

Comment:
Lambdas are not pickleable and will cause issues with the runtime system. You should define a module-level function to be used as a default here, similar to what you did for `gate_count`.